main.py

```diff
@@ -0,0 +1,319 @@
+            f"{self.format_date(extremes['lowest_temp_day'])}")
+        print(
+            f"Humidity: {extremes['highest_humidity']}% on "
+            f"{self.format_date(extremes['highest_humidity_day'])}")
+
+    @staticmethod
+    def format_date(date_str):
+        parts = date_str.split('-')
+        # year = parts[0]
+        month = parts[1]
+        day = parts[2]
+        month_name = WeatherReportsGenerator.get_month_name(int(month))
+        return f"{month_name} {day}"
+
+    @staticmethod
+    def get_month_name(month):
+        months = ["January", "February", "March", "April", "May", "June",
+                  "July", "August", "September", "October", "November",
+                  "December"]
+        return months[month - 1]
+
+    def display_monthly_averages(self, year, month):
+        averages = WeatherResultsCalculator.calculate_monthly_averages(
+            self.readings, year, month)
+        print(f"Highest Average: {averages['avg_max_temp']:.2f}C")
+        print(f"Lowest Average: {averages['avg_min_temp']:.2f}C")
+        print(f"Average Mean Humidity: {averages['avg_mean_humidity']:.2f}%")
+
+    def draw_monthly_temperature_charts(self, year, month):
+        daily_temperature_readings = {}
+
+        for reading in self.readings:
+            reading_year, reading_month, reading_day = reading.date.split('-')
+            if reading_year == year and reading_month == month:
+                day = int(reading_day)
+                if day not in daily_temperature_readings:
+                    daily_temperature_readings[day] = {'max_temp': None,
+                                                       'min_temp': None}
+                if reading.max_temp is not None:
+                    daily_temperature_readings[day]['max_temp'] = (
+                        reading.max_temp)
+                if reading.min_temp is not None:
+                    daily_temperature_readings[day]['min_temp'] = (
+                        reading.min_temp)
+
+        month_name = WeatherReportsGenerator.get_month_name(int(month))
+        print(f"{month_name} {year}")
+
+        # ANSI escape codes for color formatting
+        _RED = '\033[91m'  # Red
+        _BLUE = '\033[94m'  # Blue
+        _END_COLOR = '\033[0m'  # Reset color
+
+        for day in sorted(daily_temperature_readings.keys()):
+            max_temp = daily_temperature_readings[day]['max_temp']
+            min_temp = daily_temperature_readings[day]['min_temp']
+
+            # To calculate colored bars for maximum and minimum temperatures
+            max_bar = f"{_RED}{'+' * int(max_temp)}{_END_COLOR}" if (
+                max_temp) else ''
+            min_bar = f"{_BLUE}{'+' * int(min_temp)}{_END_COLOR}" if (
+                min_temp) else ''
+
+            print(f"{day:02} {max_bar} {max_temp}C")
+            print(f"{day:02} {min_bar} {min_temp}C")
+
+
+def main():
+    if len(sys.argv) < 4 or len(sys.argv) % 2 != 0:
```

Comment:
this function can be better made better by making smaller functions out of different function in it.